src/components/Expandable.svelte

```diff
@@ -0,0 +1,70 @@
+<script>
+  import { onMount } from 'svelte';
+  import { circInOut } from 'svelte/easing';
+
+  export let preview = null;
+  export let fullScreenComponent = null;
+
+  let parentRef = null;
+  let isExpanded = false;
+  let isFullVisible = false;
+
+  function animate(node, { duration = 600, parent }) {
+    const {
+      bottom,
+      top,
+      right,
+      left,
+      height,
+      width
+    } = parent.getBoundingClientRect();
+
+    return {
+      duration,
+      css: t => {
+        const eased = circInOut(t);
+
+        return `
+          left: ${left * (1 - eased)}px;
```

Comment:
how about to think about use native CSS animations with transition and transformation?
Then inside the component, you can only toggle classes.
Here is a tip link about this
https://medium.com/outsystems-experts/how-to-achieve-60-fps-animations-with-css3-db7b98610108

Reply:
I'll try)